core/services/blockhashstore/coordinators.go

```diff
@@ -0,0 +1,171 @@
+package blockhashstore
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/pkg/errors"
+
+	v1 "github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/solidity_vrf_coordinator_interface"
+	v2 "github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/vrf_coordinator_v2"
+)
+
+var (
+	_ Coordinator = MultiCoordinator{}
+	_ Coordinator = &V1Coordinator{}
+	_ Coordinator = &V2Coordinator{}
+)
+
+// MultiCoordinator combines the data from multiple coordinators.
+type MultiCoordinator []Coordinator
+
+// NewMultiCoordinator creates a new Coordinator that combines the results of the given
+// coordinators.
+func NewMultiCoordinator(coordinators ...Coordinator) Coordinator {
+	if len(coordinators) == 1 {
+		return coordinators[0]
+	}
+	return MultiCoordinator(coordinators)
+}
+
+// Requests satisfies the Coordinator interface.
+func (m MultiCoordinator) Requests(
+	ctx context.Context,
+	fromBlock uint64,
+	toBlock uint64,
+) ([]Event, error) {
+	var reqs []Event
+	for _, c := range m {
+		r, err := c.Requests(ctx, fromBlock, toBlock)
+		if err != nil {
+			return nil, fmt.Errorf("%w", err)
```

Comment:
Any reason we're doing `fmt.Errorf` instead of just returning `err` here?

Reply:
Ah I see. There's `errors.WithStack` but I wasn't aware the `%w` format specifier includes that info! TIL.